src/Repository.js

```diff
@@ -1,81 +1,94 @@
-import GitUtilities from "./GitUtilities";
-import FileSystemUtilities from "./FileSystemUtilities";
+import path from "path";
+import findUp from "find-up";
+import loadJsonFile from "load-json-file";
 import PackageUtilities from "./PackageUtilities";
 import Package from "./Package";
 import NpmUtilities from "./NpmUtilities";
-import path from "path";
-import logger from "./logger";
 
 const DEFAULT_PACKAGE_GLOB = "packages/*";
 
 export default class Repository {
   constructor() {
-    if (!GitUtilities.isInitialized()) {
-      logger.info("Initializing Git repository.");
-      GitUtilities.init();
-    }
+    // findUp returns null when not found, and path.resolve starts from process.cwd()
+    const lernaJsonLocation = findUp.sync("lerna.json") || path.resolve("lerna.json");
 
-    this.rootPath = path.resolve(GitUtilities.getTopLevelDirectory());
-    this.lernaJsonLocation = path.join(this.rootPath, "lerna.json");
+    this.rootPath = path.dirname(lernaJsonLocation);
+    this.lernaJsonLocation = lernaJsonLocation;
     this.packageJsonLocation = path.join(this.rootPath, "package.json");
-    this.packagesLocation = path.join(this.rootPath, "packages"); // TODO: Kill this.
-
-    // Legacy
-    this.versionLocation = path.join(this.rootPath, "VERSION");
-
-    if (FileSystemUtilities.existsSync(this.lernaJsonLocation)) {
-      this.lernaJson = JSON.parse(FileSystemUtilities.readFileSync(this.lernaJsonLocation));
-    } else {
-      // No need to distinguish between missing and empty.
-      // This saves us a lot of guards.
-      this.lernaJson = {};
-    }
+  }
 
-    if (FileSystemUtilities.existsSync(this.packageJsonLocation)) {
-      this.packageJson = JSON.parse(FileSystemUtilities.readFileSync(this.packageJsonLocation));
+  get lernaJson() {
+    if (!this._lernaJson) {
+      try {
+        this._lernaJson = loadJsonFile.sync(this.lernaJsonLocation);
+      } catch (ex) {
+        // No need to distinguish between missing and empty,
+        // saves a lot of noisy guards elsewhere
+        this._lernaJson = {};
+      }
     }
 
-    this.package = new Package(this.packageJson, this.rootPath);
+    return this._lernaJson;
   }
 
   get lernaVersion() {
-    return this.lernaJson && this.lernaJson.lerna;
+    return this.lernaJson.lerna;
   }
 
   get version() {
-    return this.lernaJson && this.lernaJson.version;
-  }
-
-  get publishConfig() {
```

Comment:
The `publishConfig` and `bootstrapConfig` properties are already handled by `Command.getOptions()`, so it seemed safe to remove these.